squidDrone/tests/com/com_interface_tests.cpp

```diff
@@ -0,0 +1,54 @@
+#include "gtest/gtest.h"
+#include "com_interface.hpp"
+
+namespace{
+  class ConcreteComInterface final: public com::ComInterface {
+    private:
+      std::array<std::uint8_t, 32> ret_array = {0};
+
+    public:
+      using com::ComInterface::msg_buffer;
+      explicit ConcreteComInterface(std::unique_ptr<com::ComMessageBuffer> com_buf) : com::ComInterface(std::move(com_buf)){}
+
+      auto get_data_packet() const noexcept -> std::array<std::uint8_t, 32>{
+        return msg_buffer->get_data();
+      }
+
+      auto put_data_packet(std::uint8_t target_id, std::array<std::uint8_t, 32> &payload) const noexcept-> std::uint8_t{
+        return 1;
+      }
+  };
+
+  class ComInterfaceTests: public ::testing::Test{
+    protected:
+      virtual void SetUp(){
+      }
+  };
+}
+
+TEST_F(ComInterfaceTests, is_constructible_with_com_buffer){
+  auto com_buffer_ = std::make_unique<com::ComMessageBuffer>();
```

Comment:
com_buffer_ and others are not private class member variables and should be written like com_buffer . They are scope locally public so there is no need to mark them as private.

Reply:
Corrected according to styleguide

Reply:
Added destructor

Reply:
Changed variable names